server/opts.go

```diff
@@ -27,13 +27,27 @@ import (
 	"regexp"
 	"strconv"
 	"strings"
+	"sync/atomic"
 	"time"
 
 	"github.com/nats-io/jwt"
 	"github.com/nats-io/nats-server/conf"
 	"github.com/nats-io/nkeys"
 )
 
+var allowUnknownTopLevelField = int32(0)
+
+// AllowUnknownTopLevelConfigurationField sets if the processing of a
+// configuration file returns an error if it finds an unknown top-level
+// configuration option.
+func AllowUnknownTopLevelConfigurationField(allow bool) {
```

Comment:
This is public and will stay forever. Maybe shorter?

```go
ErrorOnUnknownFields()
```

Reply:
Will update, no problem.

Reply:
Actually no, or change the meaning. That is, I want the default behavior to not ignore, and not have to invoke this function in normal cases. That is, I want streaming to invoke the function to change default behavior. So any other idea for the name?

Reply:
NoErrOnUnknownFields